web/src/containers/ActivityDetailAll.js

```diff
@@ -0,0 +1,35 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+import ActivityDetailDescription from './ActivityDetailDescription';
+import ActivityDetailGoals from './ActivityDetailGoals';
+import Collapsible from '../components/Collapsible';
+
+const activityTitle = (a, i) => {
+  let title = `Activity ${i}`;
+  if (a.name) title += `: ${a.name}`;
+  if (a.types.length) title += ` (${a.types.join(', ')})`;
+  return title;
+};
+
+const ActivityDetailAll = ({ aId, title }) => (
+  <Collapsible title={title} bgColor="darken-1" open>
+    <ActivityDetailDescription aId={aId} />
+    <ActivityDetailGoals aId={aId} />
+  </Collapsible>
+);
+
+ActivityDetailAll.propTypes = {
+  aId: PropTypes.number.isRequired,
+  title: PropTypes.string.isRequired
+};
+
+const mapStateToProps = ({ activities: { byId } }, { aId, num }) => {
+  const activity = byId[aId];
```

Comment:
I'm curious why you get the activity from redux store by ID rather than taking it as a param?

Reply:
yeah, i wasn't really sold on this either; i read that it was better performance to pass down the id and look up the data object via connect, let me track down the article

Reply:
https://github.com/reactjs/redux/issues/1751

Reply:
Fascinating.  Really gets into the weeds of how React handles re-rendering as state changes.